src/features/pastelAddressBlock/PastelAddressBlock.tsx

```diff
@@ -0,0 +1,292 @@
+import dateformat from 'dateformat'
+import { clipboard, shell } from 'electron'
+import QRCode from 'qrcode.react'
+import React, { useEffect, useState } from 'react'
+import {
+  AccordionItem,
+  AccordionItemButton,
+  AccordionItemHeading,
+  AccordionItemPanel,
+} from 'react-accessible-accordion'
+
+import Utils from '../../legacy/utils/utils'
+import styles from './PastelAddressBlock.module.css'
+
+type AddressBalanceProps = {
+  address: string
+  balance: number
+}
+
+type TransactionsProps = {
+  address: string
+  time: number
+  type: string
+  inputAddresses?: string[]
+}
+
+type PastelAddressBlockProps = {
+  addressBalance: AddressBalanceProps
+  label?: string
+  currencyName: string
+  pslPrice: number
+  privateKey: string
+  fetchAndSetSinglePrivKey: (address: string, type?: string) => void
+  viewKey: string
+  fetchAndSetSingleViewKey: (address: string) => void
+  rerender: any
+  transactions?: [TransactionsProps]
+}
+
+export default function PastelAddressBlock({
+  addressBalance,
+  label,
+  currencyName,
+  pslPrice,
+  privateKey,
+  fetchAndSetSinglePrivKey,
+  viewKey,
+  fetchAndSetSingleViewKey,
+  transactions,
+}: PastelAddressBlockProps): JSX.Element {
+  const { address } = addressBalance
+  const [copied, setCopied] = useState(false)
+  const [timerID, setTimerID] = useState(null)
+  useEffect(() => {
+    return () => {
+      if (timerID) {
+        clearTimeout(timerID as any)
+      }
+    }
+  })
+  const balance = addressBalance.balance || 0
+
+  const openAddress = () => {
+    if (currencyName === 'LSP') {
+      shell.openExternal(`https://chain.so/address/PSLTEST/${address}`)
+    } else {
+      shell.openExternal(`https://explorer.pastel.network/address/${address}`)
+    }
+  }
+
+  const getDifferenceDays = (date: Date) => {
+    try {
+      const now = new Date()
+      const distance = now.getTime() - date.getTime()
+      const days = Math.floor(distance / (1000 * 60 * 60 * 24))
+
+      if (days < 1) {
+        const hours = Math.floor(
+          (distance % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60),
+        )
+
+        if (hours < 1) {
+          const minutes = Math.floor(
+            (distance % (1000 * 60 * 60)) / (1000 * 60),
+          )
+
+          return `${minutes || 1} ${minutes <= 1 ? 'minute' : 'minutes'}`
+        }
+
+        return `${hours || 1} ${hours === 1 ? 'hour' : 'hours'}`
+      }
+
+      return `${days}  ${days === 1 ? 'day' : 'days'}`
+    } catch {
+      return null
+    }
+  }
```

Comment:
I think this is good, but it requires support and must be maintained. For this reason, let's use `dayjs`, and here is the example how to get formatted time from now: https://day.js.org/docs/en/display/from-now

Reply:
I have updated and used dayjs